packages/gatsby-theme/src/components/Share/ShareReddit.js

```diff
@@ -0,0 +1,24 @@
+import PropTypes from "prop-types"
+import { RedditShareButton } from "react-share"
+import Icon from "../Icon"
+
+const ShareReddit = ({ title, to, children, ...props }) => {
+  return (
+    <RedditShareButton url={to} title={title} resetButtonStyle={false} {...props}>
+      <Icon icon='reddit' />
+      {children}
+    </RedditShareButton>
+  )
```

Comment:
Same refactoring of using a single component with a Hashmap to resolve the specific component from the "react-share" library